### PR TITLE
Fix a bug in stateless re connection

### PR DESCRIPTION
**Pull request by Pascal-Verdier**
- Fix a bug in stateless re connection
- Remove warnings
- Add myself as developer in pom.xml

### CODE REVIEW

File: tangorb/dao/src/main/java/fr/esrf/TangoApi/events/EventConsumer.java [hunk 1]
```diff
@@ -53,13 +53,13 @@
         implements TangoConst, IEventConsumer {
 
     protected static int subscribe_event_id = 0;
-    protected static Hashtable<String, EventChannelStruct>   channel_map        = new Hashtable<String, EventChannelStruct>();
-    protected static Hashtable<String, String>               device_channel_map = new Hashtable<String, String>();
-    protected static Hashtable<String, EventCallBackStruct>  event_callback_map = new Hashtable<String, EventCallBackStruct>();
-    protected static Hashtable<String, EventCallBackStruct>  failed_event_callback_map = new Hashtable<String, EventCallBackStruct>();
+    protected static Hashtable<String, EventChannelStruct>   channel_map        = new Hashtable<>();
```
Comment (Ingvord) [inline tangorb/dao/src/main/java/fr/esrf/TangoApi/events/EventConsumer.java:56]: Does this really need to be static?

File: tangorb/dao/src/main/java/fr/esrf/TangoApi/events/EventConsumer.java [hunk 2]
```diff
@@ -53,13 +53,13 @@
...
+    protected static Hashtable<String, String>               device_channel_map = new Hashtable<>();
+    protected static Hashtable<String, EventCallBackStruct>  event_callback_map = new Hashtable<>();
+    protected static Hashtable<String, EventCallBackStruct>  failed_event_callback_map = new Hashtable<>();
 
     //  Alternate tango hosts
-    static ArrayList<String>  possibleTangoHosts = new ArrayList<String>();
+    static ArrayList<String>  possibleTangoHosts = new ArrayList<>();
```
Comment (Ingvord) [inline tangorb/dao/src/main/java/fr/esrf/TangoApi/events/EventConsumer.java:62]: Does this need to be static?

File: tangorb/dao/src/main/java/fr/esrf/TangoApi/events/EventConsumer.java (hunk 1 above)
Comment (Pascal-Verdier) [reply tangorb/dao/src/main/java/fr/esrf/TangoApi/events/EventConsumer.java:56]: Yes.
The same maps are used by ZMQ EventConsumer and Notifd EventConsumer.
It has been developed as static 12 years ago, why do you want to change now ?

File: tangorb/dao/src/main/java/fr/esrf/TangoApi/events/EventConsumer.java (hunk 2 above)
Comment (Pascal-Verdier) [reply tangorb/dao/src/main/java/fr/esrf/TangoApi/events/EventConsumer.java:62]: It has been developed as static 12 years ago, why do you want to change now ?

File: tangorb/dao/src/main/java/fr/esrf/TangoApi/events/EventConsumer.java (hunk 1 above)
Comment (Ingvord) [reply tangorb/dao/src/main/java/fr/esrf/TangoApi/events/EventConsumer.java:56]: @Pascal-Verdier  thanks for the clarification. Static is evil nowdays: [Why static is evil](https://stackoverflow.com/questions/7026507/why-are-static-variables-considered-evil) :) For now lets leave it as it is. We may change this once we drop support of Notifd.

File: tangorb/dao/src/main/java/fr/esrf/TangoApi/events/EventConsumer.java (hunk 2 above)
Comment (Ingvord) [reply tangorb/dao/src/main/java/fr/esrf/TangoApi/events/EventConsumer.java:62]: @Pascal-Verdier  thanks for the clarification. Static is evil nowdays: [Why static is evil](https://stackoverflow.com/questions/7026507/why-are-static-variables-considered-evil) :) For now lets leave it as it is. We may change this once we drop support of Notifd.

**Comment by Ingvord** [comment]
Released in  [ 9.3.2 ](https://bintray.com/tango-controls/maven/JTango/9.3.2/link)

**Pull request merged**